src/signature.rs

```diff
@@ -312,21 +287,19 @@ pub fn verify_messages(
     let mut valid = true;
     let mut pairing = blstrs::PairingG1G2::new(true, CSUITE);
     for (message, public_key) in messages.iter().zip(public_keys.iter()) {
-        let res = pairing.aggregate(&public_key.0.into(), None, message, &[]);
-        if res.is_err() {
-            valid = false;
-            break;
+        if pairing
```

Comment:
I'm really not sure why the previous code was written with the `valid` boolean that way.

Reply:
Yeah, it;'s definitely goofy. I'm guessing that this function was originally using combinators rather than `for` loops.